rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -386,6 +393,10 @@ private int tryDoEnd(boolean compensate) {
 
         updateStatus(compensate);
 
+        if (parentId != null && status == ParticipantStatus.Completed) {
```

Comment:
parentId is set in the constructor, so won't this be executed even when the nested LRA is completed (and parent is still active)?

Reply:
That is an excellent point. Let me go back and take a look at your original PR ...

Reply:
I added a check to test whether the close call is because the parent was asked to close.